src/snowflake/cli/_plugins/nativeapp/commands.py

```diff
@@ -249,7 +250,7 @@ def app_teardown(
         for package_entity in project.get_entities_by_type(
             ApplicationPackageEntityModel.get_type()
         ).values()
-        if package_entity.identifier == app_package_entity.identifier
+        if same_identifiers(package_entity.identifier, app_package_entity.identifier)
```

Comment:
same_identifiers apply only for strings. iirc package_entity.identifier can be of type Identifier, which includes name/schema/database.

Reply:
oof great catch. I didn't know identifier field can be a name/schema/database at all. Thank you.